src/omotes_sdk/omotes_interface.py

```diff
@@ -82,23 +90,32 @@ class OmotesInterface:
 
     broker_if: BrokerInterface
     """Interface to RabbitMQ broker."""
-    workflow_type_manager: WorkflowTypeManager
-    """Manager of all possible workflows."""
+    workflow_type_manager: WorkflowTypeManager | None
+    """All available workflow types."""
 
-    def __init__(self, rabbitmq_config: RabbitMQConfig, possible_workflows: WorkflowTypeManager):
+    def __init__(
+        self,
+        rabbitmq_config: RabbitMQConfig,
+    ):
         """Create the OMOTES interface.
 
         NOTE: Needs to be started separately.
 
         :param rabbitmq_config: RabbitMQ configuration how to connect to OMOTES.
-        :param possible_workflows: Container for all workflows which are expected to exist.
+        workflows are updated.
         """
         self.broker_if = BrokerInterface(rabbitmq_config)
-        self.workflow_type_manager = possible_workflows
+        self.workflow_type_manager = None
 
     def start(self) -> None:
-        """Start any other interfaces."""
+        """Start any other interfaces and request available workflows."""
         self.broker_if.start()
+        self.connect_to_available_workflows_updates()
+        self.request_available_workflows()
+
+        while not self.workflow_type_manager:
```

Comment:
An alternative would be to use an `threading.Event` https://docs.python.org/3/library/threading.html#event-objects . If you create an `self._workflow_received = threading.Event()` it will provide you with the option to `self._workflow_received.wait(timeout=5)` as an alternative to sleeping. And, you could change the `while not self.workflow_type_manager` condition to `while not self._workflow_received.is_set()`. Finally, you would set the event when the workflows are received at at the end of the function `def callback_on_update_available_workflows` by adding `self._workflow_received.set()`.

`threading.Event` is thread-safe which means it is possible to manipulate this object from multiple threads at the same time. The `callback` and `start` functions are running in different threads. Current implementation will work as setting a variable is atomic in Python, but it would only check if the workflows are set every 5 seconds. Using `threading.Event` allows you to break the `while`-loop the moment the workflows become available.